scripts/SANS/sans/gui_logic/models/batch_process_runner.py

```diff
@@ -65,7 +77,18 @@ def _process_states_on_thread(self, states, use_optimizations, output_mode, plot
             except Exception as e:
                 self.row_failed_signal.emit(key, str(e))
 
-    def _load_workspaces_on_thread(self, states):
+    def _load_workspaces_on_thread(self, selected_rows, get_states_func, get_thickness_for_rows_func):
+        get_thickness_for_rows_func()
+        # The above must finish before we can call get states
+        states, errors = get_states_func(row_index=selected_rows)
+
+        for row, error in errors.items():
+            self.row_failed_signal.emit(row, error)
+
+        if not states:
+            self.row_processed_signal.emit(None)
```

Comment:
Elsewhere this signal is emitted with 3 arguments. Will it work being emitted with one here?